hashes/src/util.rs

```diff
@@ -15,22 +15,20 @@
 #[macro_export]
 /// Adds hexadecimal formatting implementation of a trait `$imp` to a given type `$ty`.
 macro_rules! hex_fmt_impl(
-    ($ty:ident) => (
-        $crate::hex_fmt_impl!($ty, );
+    ($ty:ident, $len:expr) => (
+        $crate::hex_fmt_impl!($ty, $len, );
     );
-    ($ty:ident, $($gen:ident: $gent:ident),*) => (
+    ($ty:ident, $len:expr, $($gen:ident: $gent:ident),*) => (
         impl<$($gen: $gent),*> $crate::_export::_core::fmt::LowerHex for $ty<$($gen),*> {
             fn fmt(&self, f: &mut $crate::_export::_core::fmt::Formatter) -> $crate::_export::_core::fmt::Result {
                 #[allow(unused_imports)]
-                use $crate::{Hash as _, HashEngine as _, hex};
+                use $crate::Hash as _;
+                use bitcoin_internals::hex::Case;
 
-                if f.alternate() {
-                    write!(f, "0x")?;
-                }
                 if $ty::<$($gen),*>::DISPLAY_BACKWARD {
-                    hex::format_hex_reverse(&self.0, f)
+                    bitcoin_internals::fmt_hex_exact!(f, $len, self.0.iter().rev(), Case::Lower)
```

Comment:
This is not correct because it means users of `hash_newtype` (which is a public macro) are required to depend on `bitcoin_internals`?

Reply:
Is this not an argument for `hex` being a crate, `hashes` depending on it and re-exporting it?